File: LogicalThinkingPatterns/SymmetricButterflyPattern/SymmetricButterfly.ts
class SymmetricButterfly {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = 0; j <= i; j++) {
        pattern += "* ";
      }
      for (let k = 4 * (n - i - 1); k > 0; k--) {
        pattern += " ";
      }
      for (let l = 0; l <= i; l++) {
        pattern += "* ";
      }
      console.log(pattern);
    }

    for (let i = n - 1; i > 0; i--) {
      let pattern: string = "";
      for (let j = 0; j < i; j++) {
        pattern += "* ";
      }
      for (let k = 0; k < 4 * (n - i); k++) {
        pattern += " ";
      }
      for (let l = 0; l < i; l++) {
        pattern += "* ";
      }
      console.log(pattern);
    }
  }
}

const butterfly = new SymmetricButterfly();
butterfly.printPattern(5);

File: LogicalThinkingPatterns/SymmetricButterflyPattern/SymmetricButterfly.py
class SymmetricButterfly:
    def printPattern(self, n):
        for i in range(0, n):
            pattern = ""
            for j in range(0, i+1):
                pattern += "* "
            for k in range(4 * (n-i-1), 0, -1):
                pattern += " "
            for l in range(0, i+1):
                pattern += "* "
            print(pattern)
        
        for i in range(n-1, 0, -1):
            pattern =""
            for j in range(0,i):
                pattern += "* "
            for k in range(0, 4*(n-i)):
                pattern += " "
            for l in range(0,i):
                pattern +="* "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

symmetricButterfly = SymmetricButterfly()
symmetricButterfly.run()